login_server/player/settings.py

```diff
@@ -37,6 +39,18 @@ def is_eligible_for_first_win(self) -> bool:
         next_eligible_time = datetime.datetime.combine(next_eligible_date, datetime.time(0, 0, 0))
         return datetime.datetime.utcnow() > next_eligible_time
 
+    def earn_xp(self, time_played: int, was_win: bool) -> None:
+        # Base XP (purely from time played)
+        xp_earned = time_played * BASE_XP_PER_SECOND
```

Comment:
This we need to cap to the time on the server.
And now that I think about it, do you only earn xp if you stay until the end of the match?

Reply:
On official servers you still earn XP on match end, even if you left partway through. I believe that behaviour will be the same here since the controller remembers the player replication infos of players who have left.

And yeah, I'll add a cap based on the server uptime before we merge this!

Reply:
Have added logic for capping, the required tamods-server DLL is also available now.